src/providers/haskell.rs

```diff
@@ -0,0 +1,76 @@
+use std::collections::BTreeMap;
+
+use serde::Deserialize;
+
+use crate::nixpacks::{
+    nix::Pkg,
+    phase::{BuildPhase, InstallPhase, SetupPhase, StartPhase},
+};
+
+use super::Provider;
+
+pub struct HaskellStackProvider {}
+
+impl Provider for HaskellStackProvider {
+    fn name(&self) -> &str {
+        "haskell_stack"
+    }
+
+    fn detect(
+        &self,
+        app: &crate::nixpacks::app::App,
+        _env: &crate::nixpacks::environment::Environment,
+    ) -> anyhow::Result<bool> {
+        Ok(app.includes_file("package.yaml"))
```

Comment:
I think this should check for something more than a single yaml file to check if the app is Haskell based. What about checking for `.cabal` or `.hs` files?

Reply:
I was originally thinking that there could be another provider for directly Cabal-based projects. This is specifically for Stack projects, and Stack uses `package.yaml`. I was originally also looking for `stack.yaml`, but it looks like that’s not actually necessarily always going to be there.

Reply:
That makes sense. I think checking for the existing of a `.hs` file would also be useful here too. Only because `package.yaml` is not Haskell specific could be used for another language.

Reply:
Fixed! It now looks for `package.yaml`, plus anything with a `.hs` extension anywhere in the source tree.